### ISSUE TITLE
WinCE port?

### ISSUE THREAD

**Issue by roytam1** [RamonUnch/GreenPad#87]
I think it could be fun to have a WinCE port.
(although I don't have actual hardware for testing now, I used to have Dell x50v in the past but it is no longer in operable state)

for a emulated environment, there is https://github.com/Milek7/wince-qemu

**Comment by RamonUnch** [comment]
True, I thought about WinCE support, Emulation will probably be the only way to test it for me too. I never had a WinCE machine, I was not even aware that it supported x86.

The on real hardware I only have Win7/XP/2000/ME/98SE/95/NT3.51 and Win3.11fw + Win32s1.30c which is not bad already.

**Comment by renodr** [comment]
Is there still interest in this? I've got more experience than any reasonable person should have with Windows CE and I could probably cook something up. I have SDKs and toolchains for CE 2.11, 5, 6, 7, and 2013. For 2.11-6 I generally use simulators, and then for CE 7 I use Qemu using that BSP as described above. For 2013 I just use Virtual PC on a Windows 7 machine, but it can be run in Hyper-V as well.

Because of the complexity though you'd probably have to ping be to make binaries before a release. I'd happily put the project files and any changes that are needed in a PR though for adding support.

For 2.11, it's recommended to use Windows NT 4 as a build environment.
For 5, I generally use Windows 2000
For 6, I generally use Windows XP
For 7, I generally use either Vista or Windows 7.
For 2013, I use Windows 7.

For CE 2.11 and 5 you generally would use Platform Builder to get yourself a simulator image, and then use the SDK from your Platform Builder image (or the Standard SDK!) along with Embedded Visual C++ 4.0 (note that it's totally different from *standard* Visual C++ 4.0) to compile your application. CE 6, 7, and 2013 all allow the use of Visual Studio to do everything from building your image for emulation to compiling your application.

Windows CE 3.0 and 4.0 also exist but I don't have much experience with those. Depending on whether or not there's interest though that is totally something I can look at. There are emulator ROMs on Archive.org as well as the Platform Builder for CE 4.2's emulation edition, and the same thing for CE 3.0 though that's an evaluation edition and seems to be incomplete.

Windows CE 1.0 would be another NT4-based build environment, but that version is strange. It has simulators available, but if I remember correctly it's designed for Handheld PC based devices which have small landscape screens.

If you have any interest in me doing this, let me know and I can work on it over the next couple weeks, and let me know what versions you'd like me to target. I could probably get away with binaries for various architectures, but can only truly test on the x86 platform and whatever the simulators give out. That being said though CE is said to use the exact same codebase on every single supported platform... :)

**Comment by RamonUnch** [comment]
I would be very interested to see GreenPad work on Windows CE, but as I said I do not have any real ways to test it, so It really depends on you and what you want to do.
If you do make a WinCE compatible version for sure link it here and share your changes.

I would suggest to go back in time step by step if you wanted to target older WinCE versions, I guess that little adjustments will be needed is you try to target WinCE 7. A lot more would be required to target WinCE 2.

I also would suggest to stick to x86 if this what you can test. It is better not to produce binaries that you would be unsure actually work.
My abilities to test are limited to the PC platform and I know almost nothing about WinCE.
So it is your call...
